tests/foreman/cli/test_syncplan.py

```diff
@@ -273,3 +319,271 @@ def test_verify_bugzilla_1261122(self):
         new_sync_plan = self._make_sync_plan()
         result = SyncPlan.info({'id': new_sync_plan['id']})
         self.assertIsNotNone(result.get('enabled'))
+
+    # This Bugzilla bug is private. It is impossible to fetch info about it.
+    @stubbed('Unstab when BZ1279539 is fixed')
+    @tier1
+    def test_verify_bugzilla_1279539(self):
+        """Verify product won't get synced immediately after adding association
+        with a sync plan which has already been started
+
+        @Feature: Sync Plan
+
+        @Assert: Repository was not synchronized
+
+        @BZ: 1279539
+        """
+        sync_plan = self._make_sync_plan({
+            'enabled': 'true',
+            'organization-id': self.org['id'],
+            'sync-date': datetime.utcnow().strftime("%Y-%m-%d %H:%M:%S"),
+        })
+        product = make_product({'organization-id': self.org['id']})
+        repo = make_repository({'product-id': product['id']})
+        Product.set_sync_plan({
+            'id': product['id'],
+            'sync-plan-id': sync_plan['id'],
+        })
+        with self.assertRaises(AssertionError):
+            self.validate_repo_content(
```

Comment:
Well, to ensure product is not (and won't become) synchronized i'm waiting here for some amount of time, which is usually enough for synchronization, and then check product's sync status. The question is, how many time do we need to wait? I've used default 5 minutes just in case, but your point is good, in other tests 30-90 sec is usually enough for sync to finish. The only thing is we need to keep in mind performance of environments may differ and some slower machine will require more time. You are probably more expirienced in product synchronizations than me, what do you think, how many time is considered enought? Can we decrease this value to, for example, 3 minutes?